### DISCUSSION TITLE
Typing Hits in version 7

### DISCUSSION THREAD

**Discussion by Sembauke** [algolia/instantsearch#6419] [Q&A] [answered]
Currently in version 6 hit is a directly exported member from `react-instantsearch-core`. But in version 7 it is not. How do I type `Hit` in my interface in version 7?

```ts
interface SuggestionProps {
  hit: Hit;
  handleMouseEnter: (e: React.SyntheticEvent<HTMLElement, Event>) => void;
  handleMouseLeave: (e: React.SyntheticEvent<HTMLElement, Event>) => void;
}
```

**Comment by dhayab** [comment, marked as answer]
Hi, you'll be able to import this type from `instantsearch.js` like this:

```ts
import type { Hit } from 'instantsearch.js';

interface SuggestionProps {
  hit: Hit;
 /* ... */
}
```

It also is a generic type, so you can enrich it with your records type definition. You can find an example on our [React InstantSearch getting-started code sandbox](https://codesandbox.io/p/sandbox/github/algolia/instantsearch/tree/master/examples/react/getting-started?file=/src/App.tsx:78,17).

**Reply by dhayab** [reply]
This library is a direct dependency of `react-instantsearch@7` now so it shouldn't cause this.

Some package managers (pnpm for example) require all dependencies to be explicitly declared in package.json.

Could you try checking this and that `instantsearch.js` is in your node_modules directory?

**Reply by Sembauke** [reply]
Yes, I am using pnpm. I have resolved the issue. It's a bit awkward how this works.
